aspects/schema.go

```diff
@@ -211,6 +246,66 @@ func (s *StorageSchema) getUserType(ref string) (*userTypeRefParser, error) {
 	return nil, fmt.Errorf("cannot find user-defined type %q", ref)
 }
 
+type alternativesSchema struct {
+	// schemas holds schemas for the types allowed for the corresponding value.
+	schemas []Schema
+}
+
+// Validate that raw matches at least one of the schemas in the alternative list.
+func (v *alternativesSchema) Validate(raw []byte) error {
+	var errs []error
+	for _, schema := range v.schemas {
+		err := schema.Validate(raw)
+		if err == nil {
+			return nil
+		}
+
+		errs = append(errs, err)
+	}
+
+	var sb strings.Builder
+	for _, err := range errs {
+		sb.WriteString("\n\t")
+
+		if verr, ok := err.(*ValidationError); ok {
```

Comment:
Is this `if` ever expected to not be run? If so, should an `else` add whatever `err.Error()` might be to `sb`?

Reply:
Not usually but it should be there. I removed it by accident when changing this to deal with the path prefixes in errors. Thanks for spotting it

Reply:
The line after the `if` already adds the error message whether it's a validation error or not. Initially I thought I had removed the else by mistake but it was just refactored so that the same WriteString(err) works for both cases